### ISSUE TITLE
Authfile does not support filenames containing spaces

### ISSUE THREAD

**Issue by Rbn3D** [xrootd/xrootd#1445]
I'm trying to add read/listing access to certain files on a Auth file Database, and I'm having trouble when filename (or path) contains an space character.

I've tried enclosing the whole path with double quote ` " `, with single quotes ` ' `, and also replacing spaces with `"\ "`, but XRootD fails at initialize when having these paths... It seems there is no way to specify them.

My XRootD config file:

```
oss.localroot /data
xrootd.export /


xrootd.seclib /usr/lib64/libXrdSec.so
sec.protocol pwd -d:0 -a:0 -vc:1

ofs.authorize
acc.authdb /etc/xrootd/Authfile
acc.authrefresh 60

oss.defaults readonly

```

And example of a line in Authfile that causes this issue:

```
u anon "/100/path/New Folder" rl
```
or
```
u anon '/100/path/New Folder' rl
```

xrootd.log:

```
210421 16:09:59 760032 acc_Config: Authorization system using configuration in /etc/xrootd/xrootd-default.cfg
=====> acc.authdb /etc/xrootd/Authfile
=====> acc.authrefresh 60
Config 2 authorization directives processed in /etc/xrootd/xrootd-default.cfg
210421 16:09:59 760032 acc_ConfigXeq: Missing template - "/100/path/New
Config 2 auth entries processed in /etc/xrootd/Authfile
------ Authorization system initialization failed.
++++++ Checkpoint initialization started.
Config warning: checkpoints disabled because the checkpoint directory is rooted in '/tmp'!
++++++ Checkpoint initialization completed.
------ File system server initialization failed.
210421 16:09:59 760032 XrootdConfig: Unable to load base file system using default
------ xroot protocol initialization failed.
210421 16:09:59 760032 XrdProtocol: Protocol xroot could not be loaded
------ xrootd anon@host:-1 initialization failed.

```

Thanks in advance.

**Comment by abh3** [comment]
Indeed, while there is a "spacechar" you can specify it only applies to entity names. You are the first to request that it apply to paths as well. I'll put this on the list as an enhancement request. Of course, that won't solve your immediate  problem and unfortunately I don't have an immediate solution as we never really envisioned having spaces in paths but are addressing this issue as people have complained over the years.

**Comment by abh3** [comment]
This will be addressed in release 5.4 with the "acc.encoding" directive that will allow you to specify how embedded spaces are represented. For id's the choice is a single charactert substitution as this is a safe and simple approach. For paths, it's somewhat more complicated and we decided that the safest approach is to allow the path to be url-encode (i.e. use %20 for space). The documentation will appear in the next few days. In the mean time I am closing this ticket as resolved.
